contracts/rust/deployer/src/builder.rs

```diff
@@ -383,26 +383,44 @@ impl<P: Provider + WalletProvider> DeployerArgs<P> {
             Contract::StakeTableV2 => {
                 let use_multisig = self.use_multisig;
                 let dry_run = self.dry_run;
-                let multisig_pauser = self.multisig_pauser.context(
-                    "Multisig pauser address must be set for the upgrade to StakeTableV2",
-                )?;
+                // Default to deployer address if pauser not explicitly set (for local demos)
+                let multisig_pauser = self.multisig_pauser.unwrap_or(admin);
                 let l1_client = L1Client::new(vec![self.rpc_url.clone()])?;
                 tracing::info!(?dry_run, ?use_multisig, "Upgrading to StakeTableV2 with ");
                 if use_multisig {
                     upgrade_stake_table_v2_multisig_owner(
                         provider,
                         l1_client,
                         contracts,
-                        self.rpc_url.to_string(),
-                        self.multisig.context(
-                            "Multisig address must be set when upgrading to --use-multisig flag \
-                             is present",
-                        )?,
-                        multisig_pauser,
-                        Some(dry_run),
+                        StakeTableV2UpgradeParams {
+                            rpc_url: self.rpc_url.to_string(),
+                            multisig_address: self.multisig.context(
+                                "Multisig address must be set when upgrading to --use-multisig \
+                                 flag is present",
+                            )?,
+                            pauser: multisig_pauser,
+                            dry_run: Some(dry_run),
+                        },
                     )
                     .await?;
                 } else {
+                    // Pick admin from config. StakeTable uses OpsTimelock for faster
+                    // emergency updates since it handles critical staking ops.
+                    let admin = if let Some(use_timelock_owner) = self.use_timelock_owner {
```

Comment:
since `initializeV2` takes an admin for the new `DEFAULT_ADMIN_ROLE`, the thought process is that the admin should be set as the new admin via the `upgradeToAndCall` method when `initializeV2` is called